### PR TITLE
ambient: integration test for service entry host resolution

### PR DESCRIPTION
**Pull request by kdorosh**
**Please provide a description of this PR:**

Updates ambient integration tests to use DNS capture so we can e2e test flows using SE hostnames resolved by ztunnel and sent to their final destination.

Also has some changes to our test service because the golang DNS client in the ForwardEcho service does not respect ndots; without those changes the DNS query and thus the integration test fails. Exec'ing into the pod and using regular `curl`s works fine; it's the DNS client in the golang service (prompted by e.g. `grpcurl -plaintext -d '{"method": "GET", "url": "http://serviceentry.istio.io", "forceDNSLookup": true, "newConnectionPerRequest": true }' localhost:17071 proto.EchoTestService/ForwardEcho`) that fails

cc @bleggett @dhawton

### CODE REVIEW

**Comment by kdorosh** [comment]
ztunnel dns will need something akin to https://github.com/istio/istio/pull/30647 before ci will pass here

cc @nmittler in case you already had a plan or thoughts on this

File: pilot/pkg/serviceregistry/kube/controller/ambientindex_test.go
Comment (kdorosh) [inline pilot/pkg/serviceregistry/kube/controller/ambientindex_test.go:375]: not sure how https://github.com/istio/istio/pull/45797 merged, but after it merged ci and my local tests were failing until i added the missing events

cc @nmittler @howardjohn

File: pkg/test/framework/components/echo/common/deployment/echos.go
Comment (stevenctl) [inline pkg/test/framework/components/echo/common/deployment/echos.go:263]: I thought this one usually has a sidecar?

File: tests/integration/pilot/common/traffic.go
Comment (stevenctl) [inline tests/integration/pilot/common/traffic.go:288]: LGTM, but how much coverage do we gain with this skipped?

File: pkg/test/framework/components/echo/common/deployment/echos.go
Comment (kdorosh) [reply pkg/test/framework/components/echo/common/deployment/echos.go]: I can remove this one, done in https://github.com/istio/istio/pull/45900/commits/83cbb6ba1459e533588a6a6b6a23356a3213899a

File: pilot/pkg/serviceregistry/kube/controller/ambientindex_test.go
Comment (kdorosh) [reply pilot/pkg/serviceregistry/kube/controller/ambientindex_test.go]: this has been handled by https://github.com/istio/istio/pull/45925 and thus removed after merging this with latest master

**Review by bleggett — APPROVED** [review]
LGTM, we def. want at least basic ztunnel DNS integ coverage.

File: pkg/test/framework/components/echo/cmd/echogen/testdata/golden.yaml
Comment (howardjohn) [inline pkg/test/framework/components/echo/cmd/echogen/testdata/golden.yaml:166]: as discussed offline, this is a ztunnel bug returning NOERROR instead of NXDOMAIN

Comment (kdorosh) [reply pkg/test/framework/components/echo/cmd/echogen/testdata/golden.yaml]: updated this PR to pull in https://github.com/istio/ztunnel/pull/603 now that it has merged, and removed the ndots hack

**Comment by kdorosh** [comment]
PTAL again @howardjohn @stevenctl , thank you! I removed the ndots thing, so this should be ready to go

File: tests/integration/pilot/common/traffic.go
Comment (stevenctl) [reply tests/integration/pilot/common/traffic.go:288]: I _thought_ this test was passing with the mode where we make everything get a Waypoint by default. Won't block PRs on that until CI blocks on that. 

https://github.com/istio/istio/blob/e430c81abbeba1cfb03caeff882eb4ad4ff33094/pkg/test/framework/resource/flags.go#L184-L185

**Review by stevenctl — APPROVED** [review]

**Review by howardjohn — COMMENTED** [review]
lgtm

**Pull request merged**
